src/styles/common/_buttons.scss

```diff
@@ -1,35 +1,87 @@
+@import '../constants/typography';
+@import '../constants/layout';
+@import '../mixins/buttons';
+
 /**************************************
  * Buttons
  **************************************/
+$bdl-btn-border-width: 1px;
+$bdl-btn-height: 32px;
+$bdl-btn-height-large: 40px;
+$bdl-btn-icon-size: 20px;
+$bdl-btn-text-icon-size: 16px;
+$bdl-btn-padding-horizontal: $bdl-grid-unit * 4;
+$bdl-btn-padding-vertical: 10px; // Do not reuse, use flexbox or line-height instead
 
 .btn {
     position: relative;
     display: inline-block;
     box-sizing: border-box;
-    margin: 5px;
-    padding: 10px 16px;
+    margin: 5px; // TODO: this causes problems for composition (breaking change needed to fix)
+    padding: $bdl-btn-padding-vertical $bdl-btn-padding-horizontal;
     color: $bdl-gray-80;
-    font-weight: normal;
-    font-size: 13px;
-    line-height: 10px;
+    font-weight: bold;
+    font-size: $bdl-fontSize;
+    line-height: $bdl-btn-height - 2 * $bdl-btn-border-width - 2 * $bdl-btn-padding-vertical;
     letter-spacing: .035em;
     white-space: nowrap;
     text-align: center;
     text-decoration: none;
     text-shadow: none;
     vertical-align: top;
     background-color: $white;
-    border: 1px solid $bdl-gray-30;
+    border-color: $bdl-gray-30;
     border-style: solid;
-    border-width: 1px;
-    border-radius: 4px;
+    border-width: $bdl-btn-border-width;
+    border-radius: $bdl-border-radius-med-size;
     cursor: pointer;
     -webkit-transition: border-color linear .15s, box-shadow linear .15s, background-color linear .15s;
     transition: background-color .05s ease-in-out, border-color .05s ease-in-out;
     -webkit-appearance: none;
     zoom: 1;
     -webkit-font-smoothing: antialiased;
 
+    &.bdl-is-large {
+        // Instead of using line-height + padding + border = height like the base size,
+        // just set the line-height + border; this already handles vertical alignment.
+        padding: 0 $bdl-btn-padding-horizontal;
+        font-size: $bdl-fontSize--dejaBlue;
+        line-height: $bdl-btn-height-large - 2 * $bdl-btn-border-width;
+    }
+
+    &.bdl-has-icon {
+        display: inline-flex;
+        // Buttons with icons use flexbox to align content instead of line-height/padding
+        align-items: center;
+        justify-content: center;
+        height: $bdl-btn-height;
+        padding: 0; // Content provides padding
+        line-height: 0;
+
+        .btn-content {
+            margin-left: $bdl-btn-padding-horizontal;
+        }
+
+        .bdl-btn-icon {
+            // Icon without text:
+            margin: ($bdl-btn-height - $bdl-btn-icon-size) / 2 - $bdl-btn-border-width;
+
+            // Icon with text:
+            & + .btn-content {
+                margin-right: 3 * $bdl-grid-unit;
+                margin-left: $bdl-grid-unit * 2;
+            }
+        }
+
+        &.bdl-is-large {
+            height: $bdl-btn-height-large;
```

Comment:
Should this be in the base `&.bdl-is-large` style rules above? It seems odd to only have it set for icons.

Reply:
It's needed here for specificity, and not needed above because the non-icon buttons define their heights using lineheight/padding so the height prop is redundant and conflicting. Could possibly refactor this to make it more intuitive.